upgrade/scripts/k8s/upgrade_k8s_steps.sh

```diff
@@ -0,0 +1,129 @@
+else
+  echo "INFO Starting the kubernetes upgrade to v1.29"
+  /usr/share/doc/csm/upgrade/scripts/k8s/upgrade_k8s.sh -v "1.27.16 1.28.15 1.29.15"
+  if [[ $? -ne 0 ]]; then
+    echo "ERROR Failed to upgrade kubernetes to v1.29"
+    exit 1
+  else
+    touch "$DONE_DIR/upgrade_k8s_1_29.done"
+    if [[ $? -ne 0 ]]; then
+      echo "ERROR Failed to create done file for v1.29 upgrade"
+      exit 1
+    fi
+    echo "INFO Successfully upgraded kubernetes to v1.29"
+  fi
+fi
+
+if [[ -f "$DONE_DIR/deploy_charts_post_k8s_upgrade.done" ]]; then
+  echo "INFO deploy manifests for v1.29 already completed, skipping."
+else
+  echo "INFO Deploying manifests for v1.29"
+  /usr/share/doc/csm/upgrade/scripts/k8s/deploy_charts_post_k8s_upgrade.sh
+  if [[ $? -ne 0 ]]; then
+    echo "ERROR Failed to deploy manifests for v1.29"
+    exit 1
+  else
+    touch "$DONE_DIR/deploy_charts_post_k8s_upgrade.done"
+    if [[ $? -ne 0 ]]; then
+      echo "ERROR Failed to create done file for v1.29 charts deployment"
+      exit 1
+    fi
+    echo "INFO Successfully deployed manifests for v1.29"
+  fi
+fi
+
+if [[ -f "$DONE_DIR/upgrade_k8s_1_32.done" ]]; then
+  echo "INFO kubernetes upgrade to v1.32 already completed, skipping."
+else
+  echo "INFO Starting the kubernetes upgrade to v1.32"
+  /usr/share/doc/csm/upgrade/scripts/k8s/upgrade_k8s.sh -v "1.30.12 1.31.8 1.32.5"
+  if [[ $? -ne 0 ]]; then
+    echo "ERROR Failed to upgrade kubernetes to v1.32"
+    exit 1
+  else
+    touch "$DONE_DIR/upgrade_k8s_1_32.done"
+    if [[ $? -ne 0 ]]; then
+      echo "ERROR Failed to create done file for v1.32 upgrade"
+    fi
+    echo "INFO Successfully upgraded kubernetes to v1.32"
+  fi
+fi
+
+if [[ -f "$DONE_DIR/cleanup_bss.done" ]]; then
+  echo "INFO BSS cleanup already completed, skipping."
+else
+  echo "INFO Remove upgrade and upgrade_version file creation from BSS for masters and workers."
+  /usr/share/doc/csm/upgrade/scripts/upgrade/cleanup.sh
+  if [[ $? -ne 0 ]]; then
+    echo "ERROR Failed to update BSS to remove upgrade and upgrade_version."
+    exit 1
+  else
+    touch "$DONE_DIR/cleanup_bss.done"
+    if [[ $? -ne 0 ]]; then
+      echo "ERROR Failed to create done file for BSS cleanup"
+      exit 1
+    fi
+    echo "INFO Successfully updated BSS to remove upgrade and upgrade_version."
+  fi
+fi
+
+echo "INFO Deploying manifests for v1.32"
+
+if [[ -f "$DONE_DIR/deploy_manifests.done" ]]; then
+  echo "INFO Manifests deployment already completed, skipping."
+else
+  pushd "${CSM_ARTI_DIR}" || {
+    echo "ERROR Failed to change directory to ${CSM_ARTI_DIR}"
+    exit 1
+  }
+  ./upgrade.sh
+  if [[ $? -ne 0 ]]; then
+    echo "ERROR Failed to deploy manifests for v1.32"
+    exit 1
+  else
+    echo "INFO Successfully deployed manifests for v1.32"
+    touch "$DONE_DIR/deploy_manifests.done"
+    if [[ $? -ne 0 ]]; then
+      echo "ERROR Failed to create done file for v1.32 manifests deployment"
+      exit 1
+    fi
+
+  fi
```

Comment:
Remove superfluous `else` clause to reduce indentation

```suggestion
  fi
  echo "INFO Successfully deployed manifests for v1.32"
  touch "$DONE_DIR/deploy_manifests.done"
  if [[ $? -ne 0 ]]; then
    echo "ERROR Failed to create done file for v1.32 manifests deployment"
    exit 1
  fi
```